pkg/virt-handler/cgroup/dedicated_cpus.go

```diff
@@ -0,0 +1,79 @@
+package cgroup
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/go-ps"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+)
+
+func GetDedicatedCpuCgroupManager(vmi *v1.VirtualMachineInstance) (Manager, error) {
+	// Find dedicated cgroup slice and make a manager for it
+	dedicatedCgroupSleepTime := services.CalculateBigUniqueValueForVmi(vmi)
+
+	procs, err := ps.Processes()
+	if err != nil {
+		return nil, err
+	}
+
+	var sleepProcPid int
+
+	for _, proc := range procs {
+		cmdline, err := os.ReadFile(filepath.Join("/", "proc", strconv.Itoa(proc.Pid()), "cmdline"))
+		if err != nil {
+			return nil, err
+		}
+
+		if strings.Contains(string(cmdline), fmt.Sprintf("%d", dedicatedCgroupSleepTime)) {
+			sleepProcPid = proc.Pid()
+			log.Log.V(detailedLogVerbosity).Infof("dedicated pid found: %d", proc.Pid())
+			break
+		}
+	}
+
+	if sleepProcPid == 0 {
+		return nil, fmt.Errorf("sleep proc pid is not found")
+	}
+
+	dedicatedCpusCgroupManager, err := NewManagerFromPid(sleepProcPid)
+	if err != nil {
+		return dedicatedCpusCgroupManager, err
+	}
+
+	return dedicatedCpusCgroupManager, nil
+}
+
+func getQemuKvmPid(computeCgroupManager Manager) (int, error) {
+	qemuKvmFilter := func(s string) bool { return strings.Contains(s, "qemu-kvm") }
```

Comment:
Please do not hardcode `qemu-kvm` name, since it is specific to centos/fedora/rhel. This will break downstream builds based on custom images.

Reply:
Thank you for that insight!
What is the name for other distros?
Is there a fixed list of possible names that others use, or do you think we can't rely on a specific name at all?
And if so - do you have any suggestion on how it should be handled?

Reply:
Well, the upstream qemu project uses the naming `qemu-system-${arch}` and some distros follow that. However, I am not very confident that searching the process list is a reliable approach. When libvirt starts qemu, it writes its PID in a file.  Could we maybe use here smth similar to https://github.com/kubevirt/kubevirt/commit/492c5cc158 ?

Reply:
Hey @vasiliy-ul!
The problem is that virt-launcher writes the PID from the container's perspective, and not from the host's perspective.

Turns out that our detector, which returns an isolationResult, also has a `GetQEMUProcess()` method [1]. But even there, we're searching by the executable name [2].

I think I will use the detector here, and we can think on how to improve it in the future.

[1] https://github.com/kubevirt/kubevirt/blob/v0.59.0-rc.1/pkg/virt-handler/isolation/isolation.go#L151
[2] https://github.com/kubevirt/kubevirt/blob/v0.59.0-rc.1/pkg/virt-handler/isolation/detector.go#L188

Reply:
In second thought there's no reason to use the detector here for performance reasons.
I will simply use the same list here.